code/modules/mob/living/carbon/carbon.dm

```diff
@@ -515,6 +515,10 @@
 	if(((maxHealth - total_burn) < HEALTH_THRESHOLD_DEAD) && stat == DEAD )
 		become_husk("burn")
 	med_hud_set_health()
+	if(stat == SOFT_CRIT)
+		add_movespeed_modifier(MOVESPEED_ID_CARBON_SOFTCRIT, TRUE, legacy_slowdown = SOFTCRIT_ADD_SLOWDOWN)

```

Comment:
good job you reinvented traits

Reply:
basically you've got a list of id based stuff on the mob that apply some values to it, how is this any different from traits?

Reply:
Traits are just a binary is it there or not, with a list of things giving something the traits instead of just a flag with no source information. movespeed modifiers currently store priority, legacy slowdown, and flags, and I don't feel bloating traits with this is a good idea when nothing else in traits would need complex data at this point.